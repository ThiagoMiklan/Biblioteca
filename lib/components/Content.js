"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Content = function Content(props) {
  (0, _type_validations.validate)(props, props_obj, "Content");
  var definition = props.definition == undefined ? "content" : "content " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.children);
};

var _default = Content;
exports.default = _default;