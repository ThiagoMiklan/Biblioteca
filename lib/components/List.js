"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  itens: "array",
  definition: "string"
};

var List = function List(props) {
  (0, _type_validations.validate)(props, props_obj, "List");
  return assembleList(props);
};

function assembleList(props) {
  var list = props.itens;
  var list_tags = "";

  if (list != undefined && Array.isArray(list)) {
    list_tags = /*#__PURE__*/_react.default.createElement("ol", {
      className: props.definition
    }, assembleRow(list));
  }

  return list_tags;
}

function assembleRow(list) {
  return list.map(function (item) {
    return /*#__PURE__*/_react.default.createElement("li", (0, _extends2.default)({
      key: item["key"],
      onClick: item["onClick"]
    }, item["custom"]), item["value"]);
  });
}

var _default = List;
exports.default = _default;