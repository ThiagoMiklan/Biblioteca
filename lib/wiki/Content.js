"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _react = _interopRequireWildcard(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

require("bulma/css/bulma.css");

var _Content = _interopRequireDefault(require("./components/Content"));

_reactDom.default.render( /*#__PURE__*/_react.default.createElement("div", {
  className: "container is-fluid"
}, /*#__PURE__*/_react.default.createElement(_Content.default, {
  definition: "is-small"
}, /*#__PURE__*/_react.default.createElement("h1", null, "Hello World"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque."), /*#__PURE__*/_react.default.createElement("h2", null, "Second Level"), /*#__PURE__*/_react.default.createElement("p", null, "Curabitur accumsan turpis pharetra ", /*#__PURE__*/_react.default.createElement("strong", null, "augue tincidunt"), "blandit. Quisque condimentum maximus mi, sit amet commodo arcu rutrum id. Proin pretium urna vel cursus venenatis. Suspendisse potenti. Etiam mattis sem rhoncus lacus dapibus facilisis. Donec at dignissim dui. Ut et neque nisl."), /*#__PURE__*/_react.default.createElement("ul", null, /*#__PURE__*/_react.default.createElement("li", null, "In fermentum leo eu lectus mollis, quis dictum mi aliquet."), /*#__PURE__*/_react.default.createElement("li", null, "In fermentum leo eu lectus mollis, quis dictum mi aliquet."), /*#__PURE__*/_react.default.createElement("li", null, "Aliquam nec felis in sapien venenatis viverra fermentum nec lectus."), /*#__PURE__*/_react.default.createElement("li", null, "Ut non enim metus."))), /*#__PURE__*/_react.default.createElement(_Content.default, {
  definition: "is-medium"
}, /*#__PURE__*/_react.default.createElement("h1", null, "Hello World"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque."), /*#__PURE__*/_react.default.createElement("h2", null, "Second Level"), /*#__PURE__*/_react.default.createElement("p", null, "Curabitur accumsan turpis pharetra ", /*#__PURE__*/_react.default.createElement("strong", null, "augue tincidunt"), "blandit. Quisque condimentum maximus mi, sit amet commodo arcu rutrum id. Proin pretium urna vel cursus venenatis. Suspendisse potenti. Etiam mattis sem rhoncus lacus dapibus facilisis. Donec at dignissim dui. Ut et neque nisl."), /*#__PURE__*/_react.default.createElement("ul", null, /*#__PURE__*/_react.default.createElement("li", null, "In fermentum leo eu lectus mollis, quis dictum mi aliquet."), /*#__PURE__*/_react.default.createElement("li", null, "In fermentum leo eu lectus mollis, quis dictum mi aliquet."), /*#__PURE__*/_react.default.createElement("li", null, "Aliquam nec felis in sapien venenatis viverra fermentum nec lectus."), /*#__PURE__*/_react.default.createElement("li", null, "Ut non enim metus."))), /*#__PURE__*/_react.default.createElement(_Content.default, {
  definition: "is-large"
}, /*#__PURE__*/_react.default.createElement("h1", null, "Hello World"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla accumsan, metus ultrices eleifend gravida, nulla nunc varius lectus, nec rutrum justo nibh eu lectus. Ut vulputate semper dui. Fusce erat odio, sollicitudin vel erat vel, interdum mattis neque."), /*#__PURE__*/_react.default.createElement("h2", null, "Second Level"), /*#__PURE__*/_react.default.createElement("p", null, "Curabitur accumsan turpis pharetra ", /*#__PURE__*/_react.default.createElement("strong", null, "augue tincidunt"), "blandit. Quisque condimentum maximus mi, sit amet commodo arcu rutrum id. Proin pretium urna vel cursus venenatis. Suspendisse potenti. Etiam mattis sem rhoncus lacus dapibus facilisis. Donec at dignissim dui. Ut et neque nisl."), /*#__PURE__*/_react.default.createElement("ul", null, /*#__PURE__*/_react.default.createElement("li", null, "In fermentum leo eu lectus mollis, quis dictum mi aliquet."), /*#__PURE__*/_react.default.createElement("li", null, "In fermentum leo eu lectus mollis, quis dictum mi aliquet."), /*#__PURE__*/_react.default.createElement("li", null, "Aliquam nec felis in sapien venenatis viverra fermentum nec lectus."), /*#__PURE__*/_react.default.createElement("li", null, "Ut non enim metus.")))), document.getElementById("root"));