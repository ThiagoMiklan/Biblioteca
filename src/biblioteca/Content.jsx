import React from 'react';
import getClassName from '../tools/getClassName';
import className from 'classnames';


// Componente que funciona como um tipo de container
// Fornece um componente para servir de base para outros

const Content = (props) =>{
    var definition =  className(getClassName("content "+props.definition,"Content"));
     return (
        <div class={definition}>
            {props.children}
        </div>
        );
    
}

export default Content;