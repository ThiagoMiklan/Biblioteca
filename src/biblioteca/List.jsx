// @flow
import React from 'react';
import getClassName from '../tools/getClassName';
import className from 'classnames';
import PropTypes from 'prop-types';
import checkerDefinition from '../tools/checker.js';


/*
Componente que funciona fornece a opção de criar uma lista
Não consta na documentação do Bulma um componente para Lista, sendo criado 
a partir da documentação do componente Content.
Eventos: onClick na tag <li>
Obs: Recomendável o uso do componente Content como pai do componente List, 
para evitar problemas de exibição.
*/

type Props = {
    itens?: Array<Object>,
    definition?: string
}

const List = (props:Props)=> {
        checkerDefinition(props.definition,"List");
        return (assembleList(props));    
}

 // monta o esqueleto da(s) lista(s) através do map e depois monta cada linha de cada lista
function assembleList(props) {
    var list = props.itens;
    var list_tags = "";
    
    if(list != undefined && Array.isArray(list)){
        list_tags = 
            <ol className={className(getClassName(props.definition, "List"))}>
                {assembleRow(list)}
            </ol>
        
    }
    
    return list_tags;
}

function assembleRow(list){
    return (list.map(item =><li onClick={item["onClick"]}>{item["value"]}</li> ));
}

List.propTypes ={
    definition: PropTypes.string,
    itens: PropTypes.array.isRequired

}


export default List;