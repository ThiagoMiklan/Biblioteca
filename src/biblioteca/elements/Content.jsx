import React from 'react';
import getClassName from '../../tools/getClassName';
import className from 'classnames';
import {renderToString} from 'react-dom/server'

class Content extends React.Component {

    render() {
        return (
        <div class="content">
            {assembleList(this.props)}
        </div>
        );
    }
}

 // monta o esqueleto da(s) lista(s) atrav√©s do map e depois monta cada linha de cada lista
 function assembleList(props) {
    var list = props.value;
    
    var list_tags =
        <div className="content">
            {list.map(list_item =>
                <ol className={className(getClassName(list_item["definition"], "Content"))}>
                    {assembleRow(list_item["value"])}
                </ol>
            )}
        </div>

        return list_tags;
}

function assembleRow(value){
    return value.map(item => <li>{item}</li>);
}




export default Content;