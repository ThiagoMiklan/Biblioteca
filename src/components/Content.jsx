// @flow
import * as React from 'react';
import {validate} from '../tools/type_validations.js';

type Props = {
  definition?: string,
  children?: React.Node
 }

const props_obj ={
    definition:  'string'
}

const Content = (props: Props) =>{
    validate(props, props_obj);
    var definition =  (props.definition == undefined) ? "content": "content "+ props.definition;
     return (
        <div class={definition}>
            {props.children}
        </div>
        );
    
}

export default Content;