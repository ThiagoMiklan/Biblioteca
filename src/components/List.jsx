// @flow

import React from 'react';
import { validate } from '../tools/type_validations.js';
import type {ListItem} from '../tools/types';

type Props = {
    itens?: Array<ListItem>,
    definition?: string
}

const props_obj = {
    itens: "array",
    definition: "string"
}

const List = (props:Props)=> {
    validate(props, props_obj,"List");
    return (assembleList(props));    
}

function assembleList(props) {
    var list = props.itens;
    var list_tags = "";
    
    if(list != undefined && Array.isArray(list)){
        list_tags = 
            <ol className={props.definition}>
                {assembleRow(list)}
            </ol>
        
    }
    
    return list_tags;
}

function assembleRow(list){
    return (list.map(item =><li key={item["key"]} onClick={item["onClick"]} {...item["custom"]}>{item["value"]}</li> ));
}


export default List;