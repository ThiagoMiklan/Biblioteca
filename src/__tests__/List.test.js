import React from 'react';
import Content from '../biblioteca/Content';
import List from '../biblioteca/List';
import renderer from 'react-test-renderer';

var itens = [{value:"Thiago",onClick:()=>{alert("Clicou")}}
            ,{value:"JoÃ£o",onClick:()=>{alert("Clicou")}},
             {value:"Mathias",onClick:()=>{alert("Clicou")}}]
             
test('test list', ()=>{
    const lists = renderer.create(
    <Content>
        <List definition="lower-alpha" itens={itens}></List>
        <List definition = "lower-roman" itens={itens}></List>
        <List definition = "upper-alpha" itens={itens}></List>
        <List definition = "upper-roman" itens={itens}></List>
    </Content>
    ).toJSON();  
    expect(lists).toMatchSnapshot();
});
