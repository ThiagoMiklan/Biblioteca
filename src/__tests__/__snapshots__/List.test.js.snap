// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test list 1`] = `
<div
  class="content"
>
  <ol
    className="is-lower-alpha"
  >
    <li>
      coffee
    </li>
    <li>
      tea
    </li>
    <li>
      milk
    </li>
  </ol>
  <ol
    className="is-lower-roman"
  >
    <li>
      coffee
    </li>
    <li>
      tea
    </li>
    <li>
      milk
    </li>
  </ol>
  <ol
    className="is-upper-alpha"
  >
    <li>
      coffee
    </li>
    <li>
      tea
    </li>
    <li>
      milk
    </li>
  </ol>
  <ol
    className="is-upper-roman"
  >
    <li>
      coffee
    </li>
    <li>
      tea
    </li>
    <li>
      milk
    </li>
  </ol>
</div>
`;
